var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -584,3 +539,49 @@ def headers(self):
             or None  # return None to indicate failure
 
     # For the 'libs' property - use the default handler.
+
+    @run_after('install')
+    def setup_build_tests(self):
+        """Copy the build test files after the package is installed to an
+        install test subdirectory for use during `spack test run`."""
+        self.cache_extra_test_sources('src/ksp/ksp/tutorials')
+
+    def test(self):
+        # solve Poisson equation in 2D to make sure nothing is broken:
+        spec = self.spec
+        env['PETSC_DIR'] = self.prefix
+        env['PETSC_ARCH'] = ''
+        if ('+mpi' in spec):
+            runexe = Executable(join_path(spec['mpi'].prefix.bin,
+                                          'mpiexec')).command
```

Comment:
Is this the same for all MPI implementations?

Reply:
I was hoping there was a spack universal mpiexec equivalent (from the mpi provider that spack picked up) that all mpi related packages could use. Is there a better impl of this - that I should look at?

I check superlu-dist - that had:
```
           mpiexe_f = which('srun', 'mpirun', 'mpiexec')
            if mpiexe_f:
                self.run_test(mpiexe_f.command, test_args, work_dir='.',
                              purpose='superlu-dist smoke test')
```
I wasn't sure if `which` picks up stuff from PATH. And if an incompatible mpiexec can be in PATH.

Reply:
[or if `srun` gets picked up - when `mpiexec` is more appropriate]

Reply:
I don't use MPI much. I know we have things like `spec['mpi'].mpicc` for the compiler, but I don't think we have anything for the runner. I'm guessing we probably should. @alalazo @becker33?

Reply:
@adamjstewart perhaps we can merge this in - and update when there is better common support running mpi tests.